pkg/lokifrontend/frontend/v2/frontend.go

```diff
@@ -77,8 +81,13 @@ type Frontend struct {
 
 	schedulerWorkers *frontendSchedulerWorkers
 	requests         *requestsInProgress
+
+	codec transport.Codec
 }
 
+var _ queryrangebase.Handler = &Frontend{}
+var _ transport.GrpcRoundTripper = &Frontend{}
```

Comment:
I don't understand this, is there a side effect this produces that we need?

Reply:
Ah. That's a Go pattern to assert that `*Frontend` implements `GrpcRoundTripper` and `Handler`.